pageserver/src/page_service.rs

```diff
@@ -1090,129 +1090,132 @@ impl PageServerHandler {
         let (mut handler_results, span): (
             Vec<Result<(PagestreamBeMessage, SmgrOpTimer), BatchedPageStreamError>>,
             _,
-        ) = match batch {
-            BatchedFeMessage::Exists {
-                span,
-                timer,
-                shard,
-                req,
-            } => {
-                fail::fail_point!("ps::handle-pagerequest-message::exists");
-                (
-                    vec![self
-                        .handle_get_rel_exists_request(&*shard.upgrade()?, &req, ctx)
-                        .instrument(span.clone())
-                        .await
-                        .map(|msg| (msg, timer))
-                        .map_err(|err| BatchedPageStreamError { err, req: req.hdr })],
+        ) = log_slow(batch.as_static_str(), LOG_SLOW_GETPAGE_THRESHOLD, async {
+            Ok::<_, QueryError>(match batch {
```

Comment:
This code is getting pretty gnarly. I opted for a minimally invasive change here that gets the job done, but could be convinced we should refactor this in a follow-up PR.

Reply:
Yeah ok for a hotfix but please move the handle_ dispatching code to a separate function and wrap that one with the slow log

Reply:
May as well just do it now, done. This could be cleaned up further by extracting out all of the observability stuff from the message, but will leave that for later.